pollution_prediction/views/pollutant_views.py

```
@@ -0,0 +1,176 @@
+            "inserted_ids": [str(id) for id in insert_result.inserted_ids],
+            "current_time": current_time,  # Include current time in the response
+            "centroid_processing_duration": centroid_duration,  # Duration for centroid prediction
+            "total_location_processing_duration": total_location_duration,  # Total duration for location data processing
+            "total_environment_processing_duration": total_environment_duration,  # Total duration for environment data processing
+        }), 201
+
+    @staticmethod
+    def get_data_by_confidence():
+        try:
+            # Parse the minimum confidence score from query parameters
+            min_confidence = float(request.args.get('min_confidence', 0))
```

Ruff (0.8.2)

Undefined name `request`

(F821)